### PR TITLE
testing/py-virtualenvwrapper: new aports

### PR DESCRIPTION
**Pull request by 4383**
aports for python virtualenvwrapper
https://bitbucket.org/virtualenvwrapper/virtualenvwrapper

### CODE REVIEW

File: testing/py-virtualenvwrapper/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Contributor: Herve Beraud <herveberaud.pro@gmail.com>
+# Maintainer: Herve Beraud <herveberaud.pro@gmail.com>
+pkgname=py-virtualenvwrapper
+pkgver=4.8.0
+pkgrel=0
+pkgdesc="Enhancements to virtualenv"
+url="https://bitbucket.org/virtualenvwrapper/virtualenvwrapper/"
+arch="noarch"
+license="non-free"
```
Comment (fabaff) [inline testing/py-virtualenvwrapper/APKBUILD]: According to the source is the license MIT.

File: testing/py-virtualenvwrapper/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+depends="python"
+makedepends='python2-dev python3-dev py-setuptools py-pbr py2-pbr'
+install=""
+#subpackages="py2-${pkgname#py-}:_py2 py3-${pkgname#py-}:_py3"
+subpackages=""
+source="${pkgname}-${pkgver}.tar.gz::https://bitbucket.org/virtualenvwrapper/virtualenvwrapper/get/$pkgver.tar.gz"
+builddir="$srcdir/"
+
+build() {
+	msg "Building python"
```
Comment (fabaff) [inline testing/py-virtualenvwrapper/APKBUILD]: Keep is as simple as possible. This add no real value.

File: testing/py-virtualenvwrapper/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	cd "$builddir/virtualenvwrapper-virtualenvwrapper-c9300524289e"
```
Comment (fabaff) [inline testing/py-virtualenvwrapper/APKBUILD]: `builddir=...` is there to make future updates easier. Now for every update both sections must be modified.

File: testing/py-virtualenvwrapper/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	export PBR_VERSION=$pkgver
+	python2 setup.py build || return 1
+	python3 setup.py build || return 1
+}
+
+package() {
```
Comment (fabaff) [inline testing/py-virtualenvwrapper/APKBUILD]: Please check https://wiki.alpinelinux.org/wiki/APKBUILD_examples:Python for details about creating packages for Python 2 and Python 3.

**Comment by 4383** [comment]
@fabaff last review comments applied! please review

**Comment by 4383** [comment]
Hello @nmeum what the next step?

**Comment by 4383** [comment]
@nmeum what does it mean the /label A-add ?

**Comment by jirutka** [comment]
A-add = Action: add new aport
Nothing you should worry about.

File: testing/py-virtualenvwrapper/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+install=""
+subpackages=""
+source="${pkgname}-${pkgver}.tar.gz::https://bitbucket.org/virtualenvwrapper/virtualenvwrapper/get/${pkgver}.tar.gz"
+builddir="${srcdir}/virtualenvwrapper-virtualenvwrapper-c9300524289e"
+
+build() {
+	cd "${builddir}"
+	export PBR_VERSION=${pkgver}
+	python2 setup.py build || return 1
+	python3 setup.py build || return 1
```
Comment (jirutka) [inline testing/py-virtualenvwrapper/APKBUILD]: `|| return 1` is not needed anymore, abuild runs with `set -e` now.

**Comment by jirutka** [comment]
Script [virtualenvwrapper.sh](https://bitbucket.org/virtualenvwrapper/virtualenvwrapper/src/36b8050a90192a087d1060f32083249d13d8a215/virtualenvwrapper.sh) is full of (unnecessary) bashisms, so you must also add `bash` to depends.

File: testing/py-virtualenvwrapper/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+_py() {
+	local python="$1"
+	pkgdesc="${pkgdesc} (for ${python})"
+	depends="${depends} ${python}"
+	install_if="${pkgname}=${pkgver}-r${pkgrel} ${python}"
+
+	cd "${builddir}"
+	export PBR_VERSION=${pkgver}
+	python setup.py install --prefix=/usr --root="${pkgdir}" --skip-build
```
Comment (jirutka) [inline testing/py-virtualenvwrapper/APKBUILD]: `python` is symlink to python2. You need to use `$python` here (contains python2 or python3).

There should be `$subpkgdir`, not `$pkgdir`.

File: testing/py-virtualenvwrapper/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	python2 setup.py build || return 1
+	python3 setup.py build || return 1
+}
+
+package() {
+	mkdir -p "${pkgdir}"
+}
+
+_py2() {
+	replaces="${pkgname}"
```
Comment (jirutka) [inline testing/py-virtualenvwrapper/APKBUILD]: This is not needed for new aports.

File: testing/py-virtualenvwrapper/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+pkgver=4.8.0
+pkgrel=0
+pkgdesc="Enhancements to virtualenv"
+url="https://bitbucket.org/virtualenvwrapper/virtualenvwrapper/"
+arch="noarch"
+license="MIT"
+depends="python"
+makedepends='python2-dev python3-dev py-setuptools py-pbr'
+install=""
+subpackages=""
```
Comment (jirutka) [inline testing/py-virtualenvwrapper/APKBUILD]: You have to declare subpackages. It’s all at the already mentioned wiki page: https://wiki.alpinelinux.org/wiki/APKBUILD_examples:Python.

File: testing/py-virtualenvwrapper/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,50 @@
+# Contributor: Herve Beraud <herveberaud.pro@gmail.com>
+# Maintainer: Herve Beraud <herveberaud.pro@gmail.com>
+pkgname=py-virtualenvwrapper
+pkgver=4.8.0
+pkgrel=0
+pkgdesc="Enhancements to virtualenv"
+url="https://bitbucket.org/virtualenvwrapper/virtualenvwrapper/"
+arch="noarch"
+license="MIT"
+depends="python"
```
Comment (jirutka) [inline testing/py-virtualenvwrapper/APKBUILD]: It should not depend on `python`.

**Comment by jirutka** [comment]
First I thought that I’ll fix the issues myself and merge it, but then I found out that this aport doesn’t work at all – it does not even produce subpackages…

**Comment by 4383** [comment]
@jirutka review comments applied!

**Comment by 4383** [comment]
@jirutka can you review please?

**Pull request closed without merging**